packages/steiger/src/models/config.ts

```diff
@@ -1,33 +1,81 @@
 import z from 'zod'
 import { createEvent, createStore } from 'effector'
+import { Config, ConfigObject, Plugin, Rule } from '@steiger/types'
 
-export const schema = z.object({
-  rules: z
-    .record(
-      z.enum([
-        'ambiguous-slice-names',
-        'excessive-slicing',
-        'forbidden-imports',
-        'inconsistent-naming',
-        'insignificant-slice',
-        'no-layer-public-api',
-        'no-public-api-sidestep',
-        'no-reserved-folder-names',
-        'no-segmentless-slices',
-        'no-segments-on-sliced-layers',
-        'public-api',
-        'repetitive-naming',
-        'segments-by-purpose',
-        'shared-lib-grouping',
-      ]),
-      z.literal('off'),
+export const $config = createStore<ConfigObject | null>(null)
+const setConfig = createEvent<ConfigObject>()
+$config.on(setConfig, (_state, payload) => payload)
+
+export const $rules = createStore<Array<Rule>>([])
+const setRules = createEvent<Array<Rule>>()
+$rules.on(setRules, (_state, payload) => payload)
+
+function processPlugins(config: Config) {
+  const plugins = config.filter((item) => 'ruleDefinitions' in item) as Array<Plugin>
+  const allRules = plugins.flatMap((plugin) => plugin.ruleDefinitions)
+  const ruleNames = allRules.map((rule) => rule.name)
+  const uniqueNames = new Set<string>(ruleNames)
+
+  // Check conflicts in rule names
+  if (uniqueNames.size !== allRules.length) {
+    const duplicates = ruleNames.filter((name, index) => ruleNames.indexOf(name) !== index)
+    throw new Error(
+      `Conflicting rule definitions found: ${duplicates.join(', ')}. Rules must be unique! Please check your plugins.`,
     )
-    .optional(),
-})
+  }
 
-export type Config = z.infer<typeof schema>
+  return allRules
+}
 
-export const $config = createStore<Config | null>(null)
-export const setConfig = createEvent<Config>()
+function mergeConfigObjects(config: Config) {
+  // TODO: temporary simplified implementation.
+  //  Implement handling the "files" and "ignores" globs in further updates.
+  return config.reduce((acc: ConfigObject, item) => {
+    if ('rules' in item) {
+      return { ...acc, rules: { ...acc.rules, ...item.rules } }
+    }
 
-$config.on(setConfig, (_state, payload) => payload)
+    return acc
+  }, {})
+}
+
+/**
+ * Dynamically build a validation scheme based on the rules provided by plugins.
```

Comment:
praise: wow, I didn't even think about this aspect, great idea!

Reply:
🤝 As always, I am happy to bring value to the project